### ISSUE TITLE
Page to fill out a generic survey for a program

### ISSUE THREAD

**Issue by willgearty** [learning-unlimited/ESP-Website#3229]
As is, I believe it's possible to make programs surveys that are not for teachers/students, but I don't believe there's currently any reason to fill out such surveys. We should have some page (tricky, since it needs to be program specific but not /learn or /teach, I guess) where a generic user could fill such a survey out. The example provided by Yale Splash was for moderators to give feedback about the classes that they moderated (obviously, this would be a separate issue since we currently don't have a way to assign moderators/volunteers to classes through the website).

(Also, perhaps there should be a volunteer-specific survey view page)

Requested by Yale at SplashCon 2021.

**Comment by willgearty** [comment]
I think this could also be achieved through custom forms? But the results of custom forms are not as pretty as those of surveys.

**Comment by kkbrum** [comment]
Now that we do assign moderators to classes, and they are technically teacher users, could there just be a second "teach" survey called "Moderator survey" for the use case suggested?

**Comment by DarshanCode2005** [comment]
Hi @willgearty, I noticed this issue is still open and currently in the Stable Release 17 milestone. I’d be interested in working on it if it’s still something we’d like to implement.
Given the previous discussion, would you prefer:
1> Creating a dedicated program-specific generic survey page,
2> Extending the existing survey system (e.g., adding a moderator-specific survey under “teach”),
3> Or leaning toward custom forms despite the reporting limitations?
I’m happy to propose a concrete implementation approach before starting, especially regarding routing (since it shouldn’t live under `/learn` or `/teach`). Let me know how you’d like to proceed!

**Comment by willgearty** [comment]
I think some combination of # 1 and # 2 would be good. I can see other use cases where you might want e.g., parents to fill out a survey about the program. Having a generic place where any user could fill out a survey would be good. This would ideally be linked into the existing survey builder (which itself could probably use a facelift), but wouldn't have a user type. I guess a fundamental question is whether a user would need an account to access the survey? Ideally they would have an account so they can't fill it out twice, but I can see some use cases (like the parents one), where the intended audience might not actually have accounts on the site and we don't want to introduce any friction to prevent potential survey fillers to not fill out the survey.

That got a bit ramble-y, but hopefully it helps @DarshanCode2005 ?

**Comment by DarshanCode2005** [comment]
Hi Dr. @willgearty, thanks for the clarification earlier. Before starting implementation, I wanted to confirm a couple of design details to make sure the solution aligns with what you had in mind:
1. Authentication requirement
   Should generic surveys require users to be logged in, or should they also support **anonymous submissions** (example: for parents or external participants)?
2. Duplicate submissions
   If anonymous submissions are allowed, do we need to **prevent duplicate responses**, or is that not a concern for now?
3. Survey classification
   How would you prefer generic surveys to be represented in the system?
   * a new survey type
   * allowing surveys with **no user role restriction**
   * or something like a `generic` flag
4. Access model
   Should generic surveys be **publicly accessible via a link** (ex: `/program/<program>/survey/<survey_id>`), or should they appear somewhere in the program dashboard? Thanks! Once I have clarity on these, I can propose a concrete implementation approach.

**Comment by willgearty** [comment]
1. I think it would be good to be able to support both and this could be a field that is set at the survey level.
2. I think that would be a concern for anonymous submissions. Not sure how to resolve that. Maybe cookies? For the moment, we can probably just focus on surveys that require users to be logged-in.
3. I think it could just be a new "category" option on the Survey class
4. Yes, I think some link like that would be good

**Comment by willgearty** [comment]
@DarshanCode2005 are you still working on this?

**Comment by DarshanCode2005** [comment]
Hi @willgearty Sir, I’d like to complete this task, but I’ve started working on my proposal, so it’s taking a bit more time than expected. I’ll make sure to finish this soon. Thanks for your understanding!

**Comment by DarshanCode2005** [comment]
> Hi [@willgearty](https://github.com/willgearty) Sir, I’d like to complete this task, but I’ve started working on my proposal, so it’s taking a bit more time than expected. I’ll make sure to finish this soon. Thanks for your understanding!

I will try to complete the remaining work related to this issue, before this weekend.

**Comment by willgearty** [comment]
No rush, just wanted to make sure you hadn't forgotten about it.

**Comment by willgearty** [comment]
@DarshanCode2005 what's the status on this?  If you don't have the time to work on it, we should open it up to other contributors.

**Comment by DarshanCode2005** [comment]
Hi @willgearty Sir, almost done, I just need to test if it's working fine. Will try to wrap up soon.

**Comment by willgearty** [comment]
Unassigning due to inactivity.

**Comment by sahu-SuMiT** [comment]
Hi @willgearty, I'd like to take over this issue. Based on the previous discussion, here is my concise implementation plan for V1:

Model Extension: Add a category field to the Survey model (e.g., LEARN, TEACH, GENERIC). I will also add a requires_login boolean (defaulting to True for V1) to lay the database groundwork for future anonymous submissions.

Routing: Create a new, standalone URL pattern for generic surveys outside of the role-specific namespaces: /program/<program>/survey/<survey_id>.

Access Control & Duplicate Prevention (V1): Enforce standard authentication on the new route. The backend will verify the user is logged in and query the database to redirect them if request.user has already submitted a response.

Admin Builder UI: Update the survey creation interface so admins can select the "Generic" category and easily access the generated public link.
